In-tree addon inputstream.adaptive failing to build without CMAKE_CXX_STANDARD=20 (default is 17)

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Building the in-tree addon inputstream.adaptive fails. I believe this might be because it needs a CMAKE_CXX_STANDARD of 20, it's set to 17 in: 

https://github.com/xbmc/xbmc/blob/8fca3140aad3828a8381b022f1a68c5884472fe7/tools/depends/target/Toolchain_binaddons.cmake.in#L92

Which is used to generate KodiConfig.cmake?

Compiler is GCC 14

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
inputstream.adaptive should build


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
inputstream.adaptive does not build


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->
CMAKE_CXX_STANDARD set to 20, however other add-ons will need to be rebuilt and tested that they compile.


### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. Build in-tree addon: sudo make -j$(getconf _NPROCESSORS_ONLN) -C tools/depends/target/binary-addons PREFIX=/usr/local ADDONS="inputstream.adaptive"


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:
```
/home/defiant/Applications/kodi/tools/depends/target/binary-addons/aarch64-webos-linux-gnu-release/build/inputstream.adaptive/src/utils/UrlUtils.cpp: In function ‘void {anonymous}::RemovePrefixSingleDot(std::string&)’:
/home/defiant/Applications/kodi/tools/depends/target/binary-addons/aarch64-webos-linux-gnu-release/build/inputstream.adaptive/src/utils/UrlUtils.cpp:89:11: error: ‘std::string’ {aka ‘class std::__cxx11::basic_string<char>’} has no member named ‘ends_with’
   89 |   if (url.ends_with("/."))
      |           ^~~~~~~~~
/home/defiant/Applications/kodi/tools/depends/target/binary-addons/aarch64-webos-linux-gnu-release/build/inputstream.adaptive/src/utils/UrlUtils.cpp: In function ‘std::string UTILS::URL::Join(std::string, std::string)’:
/home/defiant/Applications/kodi/tools/depends/target/binary-addons/aarch64-webos-linux-gnu-release/build/inputstream.adaptive/src/utils/UrlUtils.cpp:294:42: error: ‘std::string’ {aka ‘class std::__cxx11::basic_string<char>’} has no member named ‘ends_with’
  294 |   if (relativeUrl == ".." || relativeUrl.ends_with("/.."))
```


### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->
N/A


## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->
N/A


### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: webOS 24
 - Kodi version: current (master branch)



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/tools/depends/target/Toolchain_binaddons.cmake.in
+++ b/tools/depends/target/Toolchain_binaddons.cmake.in
@@ -88,8 +88,6 @@ set(CMAKE_C_FLAGS_DEBUG "@platform_cflags_debug@ @platform_includes@")
 set(CMAKE_CXX_FLAGS_DEBUG "@platform_cxxflags_debug@ @platform_includes@")
 set(CMAKE_CPP_FLAGS "@platform_cflags@ @platform_includes@")
 set(CMAKE_EXE_LINKER_FLAGS "@platform_ldflags@")
-# set default C++ standard version - can be overridden
-set(CMAKE_CXX_STANDARD 17)
 set(ENV{CFLAGS} ${CMAKE_C_FLAGS})
 set(ENV{CXXFLAGS} ${CMAKE_CXX_FLAGS})
 set(ENV{CPPFLAGS} ${CMAKE_CPP_FLAGS})

